presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergFileWriterFactory.java

```diff
@@ -186,9 +186,9 @@ private IcebergFileWriter createOrcWriter(
                     try {
                         return new HdfsOrcDataSource(
                                 new OrcDataSourceId(outputPath.toString()),
-                                fileSystem.getFileStatus(outputPath).getLen(),
+                                hdfsEnvironment.doAs(session.getUser(), () -> fileSystem.getFileStatus(outputPath).getLen()),
```

Comment:
Same, we don't do this for `OrcFileWriterFactory`

Reply:
Same above